dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -719,17 +722,40 @@ protected String getSelectSql( QueryFilter filter, QueryItem item, Date startDat
         }
     }
 
+    private String getFilter( String filter, QueryItem item )
+    {
+        try
+        {
+            if ( item.getValueType() == ValueType.DATETIME )
+            {
+                return DateFormatUtils.format(
+                    DateUtils.parseDate( filter,
+                        // known formats
+                        "yyyy-MM-dd'T'HH.mm",
+                        "yyyy-MM-dd'T'HH.mm.ss" ),
+                    // postgres format
+                    "yyyy-MM-dd HH:mm:ss" );
+            }
+        }
+        catch ( ParseException pe )
+        {
+            log.warn( "Unparsable date using known formats: " + filter );
```

Comment:
What if we use the same approach we already have in this class. Something like:
```
log.warn( ErrorCode.E7131.getMessage(), ex );
throw new QueryRuntimeException( ErrorCode.E7131, ex );
```
where
```
ErrorCode.E7131 = "Unparsable date/time. Unknown format"
```
Just a suggestion, so the clients can have an idea about what could be wrong.

Reply:
sounds good.